app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumCacheTest.kt

```diff
@@ -0,0 +1,166 @@
+package com.github.h3lp3rs.h3lp.forum
+
+import androidx.test.core.app.ApplicationProvider
+import com.github.h3lp3rs.h3lp.database.Databases.*
+import com.github.h3lp3rs.h3lp.database.Databases.Companion.setDatabase
+import com.github.h3lp3rs.h3lp.database.MockDatabase
+import com.github.h3lp3rs.h3lp.forum.implementation.CachedForum
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.globalContext
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.resetStorage
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class CacheForumTest {
+
+    // Useful variables
+    private lateinit var rawForum: Forum
+    private lateinit var cachedForum: Forum
+
+    @Before
+    fun setup() {
+        globalContext = ApplicationProvider.getApplicationContext()
+        resetStorage()
+        setDatabase(PREFERENCES, MockDatabase())
+        ForumCategory.mockForum()
+        rawForum = ForumCategory.root().child(ForumCategory.GENERAL.name)
+        cachedForum = CachedForum(rawForum)
+    }
+
+    private fun resetOnlineForum() {
+        ForumCategory.mockForum()
+        rawForum = ForumCategory.root().child(ForumCategory.GENERAL.name)
+        cachedForum = CachedForum(rawForum)
+    }
+
+    @Test
+    fun newPostIsAddedToCache() {
+        // Add a post
+        val postOnline = cachedForum.newPost(AUTHOR, CONTENT, isPost = true).join()

```

Comment:
as usual, Im not a fan of the .join() calls, instead of modifying all the tests, maybe just add a comment at the beginning of the tests stating that since we're using the mockDatabase which runs locally, the joins won't cause any problem

Reply:
Composing futures in this case is a big mess, and alters the understanding (and would still require join() nevertheless).
If you insist I can write a comment though :-)